pkg/csi/cinder/openstack/openstack_volumes.go

```diff
@@ -412,6 +428,36 @@ func (os *OpenStack) diskIsUsed(volumeID string) (bool, error) {
 	return false, nil
 }
 
+// getVolumeTypeByName is a wrapper around ListVolumeTypes that creates a Name filter to act as a GetByUniqueName
+// Returns the Volume Type referenced with the specified name
+func (os *OpenStack) getVolumeTypeByName(n string) (*volumetypes.VolumeType, error) {
+	var matchingVolType *volumetypes.VolumeType
+
+	opts := volumetypes.ListOpts{}
+	err := volumetypes.List(os.blockstorage, opts).EachPage(func(page pagination.Page) (bool, error) {
+		var err error
+
+		volTypes, err := volumetypes.ExtractVolumeTypes(page)
+		if err != nil {
+			return false, err
+		}
+
+		for _, volType := range volTypes {
+			if n == volType.Name {
+				matchingVolType = &volType
```

Comment:
We should probably fail if there are more than one VolumeType of a certain name, or we're non-deterministic.